dds/DCPS/XTypes/DynamicType.h

```diff
@@ -0,0 +1,50 @@
+#ifndef OPENDDS_DCPS_XTYPES_DYNAMIC_TYPE_H
+#define OPENDDS_DCPS_XTYPES_DYNAMIC_TYPE_H
+
+#include "External.h"
+#include "DynamicTypeMember.h"
+
+#include <dds/DCPS/RcHandle_T.h>
+#include <dds/DCPS/RcObject.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+namespace OpenDDS {
+namespace XTypes {
+
+class TypeDescriptor;
+
+class OpenDDS_Dcps_Export DynamicType : public OpenDDS::DCPS::RcObject
+{
+public:
+  DynamicType();
+  ~DynamicType();
+
+  void get_descriptor(TypeDescriptor& descriptor) const;
+  DCPS::String get_name() const;
+  TypeKind get_kind() const;
+  DDS::ReturnCode_t get_member_by_name(DynamicTypeMember_rch& member, const DCPS::String& name);
```

Comment:
Any method that can be made in to a `const` member function probably should be one.  The only reason I could think of to not do so would be to strictly follow the mapping from the Annex C IDL, but we're not doing that either.